### ISSUE TITLE
AttributeError: 'GraphQLWSConsumer' object has no attribute '_handler'

### ISSUE THREAD

**Issue by Byloth** [strawberry-graphql/strawberry#3259]
I struggled for a few hours with this error.  
Eventually, I understood what I was doing wrong.

*... or -at least- not **just** me!*

---

I was following the [Channels documentation](https://strawberry.rocks/docs/integrations/channels) trying to implement some **Subscription** behaviour using **Django**.  
I actually read the documentation wrong so I was trying to open a new WebSocket connection by simply instantiate it in the browser as follow:

```ts
const socket = new WebSocket("ws://localhost:8000/graphql/");
```
...and promptly every time I ran this function, the server logged this:
```
INFO:     ('172.20.0.1', 45834) - "WebSocket /graphql/" 403
INFO:     connection rejected (403 Forbidden)
INFO:     connection closed
ERROR:    Exception in ASGI application
Traceback (most recent call last):
  File "/usr/lib/python3.11/site-packages/uvicorn/protocols/websockets/websockets_impl.py", line 247, in run_asgi
    result = await self.app(self.scope, self.asgi_receive, self.asgi_send)
             ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^
  File "/usr/lib/python3.11/site-packages/uvicorn/middleware/proxy_headers.py", line 84, in __call__
    return await self.app(scope, receive, send)
           ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^
  File "/usr/lib/python3.11/site-packages/channels/routing.py", line 62, in __call__
    return await application(scope, receive, send)
           ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^
  File "/usr/lib/python3.11/site-packages/channels/routing.py", line 116, in __call__
    return await application(
           ^^^^^^^^^^^^^^^^^^
  File "/usr/lib/python3.11/site-packages/channels/consumer.py", line 94, in app
    return await consumer(scope, receive, send)
           ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^
  File "/usr/lib/python3.11/site-packages/channels/consumer.py", line 58, in __call__
    await await_many_dispatch(
  File "/usr/lib/python3.11/site-packages/channels/utils.py", line 50, in await_many_dispatch
    await dispatch(result)
  File "/usr/lib/python3.11/site-packages/strawberry/channels/handlers/base.py", line 88, in dispatch
    await super().dispatch(message)
  File "/usr/lib/python3.11/site-packages/channels/consumer.py", line 73, in dispatch
    await handler(message)
  File "/usr/lib/python3.11/site-packages/channels/generic/websocket.py", line 238, in websocket_disconnect
    await self.disconnect(message["code"])
  File "/usr/lib/python3.11/site-packages/strawberry/channels/handlers/ws_handler.py", line 116, in disconnect
    await self._handler.handle_disconnect(code)
          ^^^^^^^^^^^^^
AttributeError: 'GraphQLWSConsumer' object has no attribute '_handler'
```

<details>
<summary>Eventually, I was able to find the correct <strong>Subscription</strong> to instantiate properly the WebSocket connection.</summary>

```graphql
subscription SubscribeToChatRooms {
  joinChatRooms(
    rooms: [{ roomName: "room1" }, { roomName: "room2" }]
    user: "foo"
  ) {
    roomName
    message
    currentUser
  }
}
```
</details>

---

Now... I was wrong... Definitely!

**BUT...** Is that **really** the proper error message to log in this case?  
It -rather- seems like it crashes badly due to some bug while trying to deal with this bad behavior.

Probably, it should check if `self._handler` is actually instantiated before using it or -better- it shouldn't call `disconnect` before actually connecting.

https://github.com/strawberry-graphql/strawberry/blob/a90dea2a3053e857aa2adf685d5c4f5d39c527c4/strawberry/channels/handlers/ws_handler.py#L115-L116

---

In the future, I would prefer not to receive an unclear (and apparently serious) Sentry notification every time a bot tries to open a WebSocket connection with my server... 😉

## Upvote & Fund

- We're using [Polar.sh](https://polar.sh/strawberry-graphql) so you can upvote and help fund this issue.
- We receive the funding once the issue is completed & confirmed by you.
- Thank you in advance for helping prioritize & fund our backlog.

<a href="https://polar.sh/strawberry-graphql/strawberry/issues/3259">
<picture>
  <source media="(prefers-color-scheme: dark)" srcset="https://polar.sh/api/github/strawberry-graphql/strawberry/issues/3259/pledge.svg?darkmode=1">
  <img alt="Fund with Polar" src="https://polar.sh/api/github/strawberry-graphql/strawberry/issues/3259/pledge.svg">
</picture>
</a>

**Comment by Byloth** [comment]
> Well it wasn't intentional :P

Of course... I know! ❤️

> Please, don't get me wrong; I probably wasn't able to convey the tone of my comment correctly.
> It didn't mean to be aggressive or accusatory at all! 🥹

---

> I'd be happy to review a PR, or if you have a test/minimal reproduction I can attempt a bug fix/workaround :)

It should happen on every single app that uses Django + Strawberry + Channels... It has to.  
But, yes! I would be happy to contribute! 🙃

**Comment by DoctorJohn** [comment]
Just randomly stumbled upon this issue and realized it has (implicitly) been resolved in #3638 :) (i.e., there's no handler instance anymore that could be missing in certain scenarios).
